walletlibrary/src/main/java/com/microsoft/walletlibrary/requests/serializer/PresentationExchangeResponseBuilder.kt

```diff
@@ -26,35 +27,36 @@ internal class PresentationExchangeResponseBuilder(
         requirement: Requirement,
         verifiedIdSerializer: VerifiedIdSerializer<String>
     ) {
-        requirement.serialize(this, verifiedIdSerializer)?.let {
-            when (requirement) {
-                is PresentationExchangeRequirement -> {
-                    requirement.serialize(this, verifiedIdSerializer)?.let { rawCredential ->
-                        // try adding requirement to a group
-                        vpTokens.forEach {
-                            if (it.canIncludeInGroup(requirement)) {
-                                it.include(requirement, rawCredential)
-                                return@let
-                            }
+        when (requirement) {
+            is PresentationExchangeRequirement -> {
+                requirement.serialize(this, verifiedIdSerializer)?.let { rawCredential ->
+                    // try adding requirement to a group
+                    vpTokens.forEach {
+                        if (it.canIncludeInGroup(requirement)) {
+                            it.include(requirement, rawCredential)
+                            return@let
                         }
-                        // create a new group
-                        val exception =
-                            libraryConfiguration.identifierManager.getMasterIdentifier().toNative().map { identifier ->
-                                val group = PresentationExchangeSubmissionGroup(identifier)
-                                group.include(requirement, rawCredential)
-                                vpTokens.add(group)
-                                return@let
-                            }.exceptionOrNull()
+                    }
+                    // create a new group
+                    val exception =
+                        libraryConfiguration.identifierManager.getMasterIdentifier().toNative().map { identifier ->
+                            val group = PresentationExchangeSubmissionGroup(identifier)
+                            group.include(requirement, rawCredential)
+                            vpTokens.add(group)
+                            return@let
+                        }.exceptionOrNull()
 
-                        if (exception != null) {
-                            throw exception
-                        }
+                    if (exception != null) {
+                        throw exception
                     }
                 }
-                else -> {
-                    libraryConfiguration.logger.w("Unknown credential type ${requirement.javaClass.name} returned" +
-                            " credential data that cannot be formatted in response")
-                }
+            }
+            is GroupRequirement -> {
```

Comment:
What does this do? Isn't this only PresentationExchangeRequirement?

Reply:
GroupRequirement is explicitly added even though it contributes nothing to the response format because otherwise the serializer cannot traverse the requirements correctly.